app/src/main/kotlin/com/wire/android/migration/MigrationManager.kt

```diff
@@ -126,19 +130,28 @@ class MigrationManager @Inject constructor(
                 appLogger.d("$TAG - Step 5 - Migrating messages for ${userId.value.obfuscateId()}")
                 migrateMessages(userId, it, coroutineScope).let { failedConversation ->
                     if (failedConversation.isEmpty()) {
-                        Either.Right(Unit)
+                        Either.Right(it)
                     } else {
                         Either.Left(failedConversation.values.first())
                     }
                 }
-            }.fold({
-                when (it) {
-                    is NetworkFailure.NoNetworkConnection -> MigrationData.Result.Failure.NoNetwork
-                    else -> MigrationData.Result.Failure.Messages(it.getErrorCode().toString())
+            }
+                .map {
+                    appLogger.d("$TAG - Step 6 - Clean read messages for ${userId.value.obfuscateId()}")
```

Comment:
This step should also be applied in `migrateAccountsData`,  this `migrateSingleUser` method is used for a single user after the install, but there is another method `migrate` which is called right after the update and uses `migrateAccountsData` 😉 
Maybe we could extract the common code 🤔

Reply:
Good to know 👁️